### DISCUSSION TITLE
Bandwidth usage and limits

### DISCUSSION THREAD

**Discussion by lucy2812** [Tribler/tribler#6155] [Ideas]
- [x] I have read CONTRIBUTING.rst

##### Tribler version/branch+revision: #####
v7.9.0 Stable

I would like to monitor how much bandwidth (data usage) I have used for a month.

I would also like to preset the amount of bandwidth (data usage) to allocate/donate per month.

**Comment by ichorid** [comment]
>I would like to monitor how much bandwidth (data usage) I have used for a month.

You mean, the *total* amount of data sent through Tribler, right? Why would you want that? What's the use case?

>I would also like to preset the amount of bandwidth (data usage) to allocate/donate per month.

You can set the limits for download/upload bandwidth (MB/sec) in the Settings tab. Isn't that what you want? Or you want to be able to set hard limits for the _total_ volume of data pumped through Tribler each month (Android-style)?

**Reply by lucy2812** [reply]
> You mean, the total amount of data sent through Tribler, right?

Yes, total data volume through Tribler per month.  Or at least as close you can get to the real "total".  The volume amount can reset every month (or a user selected day of month).

> Why would you want that? What's the use case?

I have ISP bandwidth/data allocation/limits.  I would like to "donate" to the community some of my bandwidth per month, while managing the risk of exceeding my ISP allocation.  I want to set a "donation" amount.  It is analogous of donating to any organization, and I cannot afford the risk of sending a blank check.

> You can set the limits for download/upload bandwidth (MB/sec) in the Settings tab. Isn't that what you want?

No...that feature is to set the velocity (MB/sec).  I need to limit the Tribler total volume for the month so I can manage my ISP data allotment.

**Reply by ichorid** [reply]
Thanks for clarifying! We'll consider implementing this in one of the upcoming releases.
